ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx

```diff
@@ -32,45 +32,53 @@ type Props = {
 } & WithBackboneProps
 
 type State = {
-  value: any
+  propertyModel: any
 }
 
 export default hot(module)(
   withListenTo(
     class EnumComponent extends React.Component<Props, State> {
       constructor(props: Props) {
         super(props)
-        this.state = {
-          value: this.props.value,
-        }
-        this.propertyModel = new PropertyModel({
+
+        const propertyModel = new PropertyModel({
           label: this.props.label,
           enum: this.props.options,
-          value: [this.state.value],
+          value: [this.props.value],
           isEditing: true,
         })
+
+        this.state = { propertyModel: propertyModel }
       }
-      propertyModel: any
       componentDidMount() {
         if (this.props.onChange !== undefined) {
           this.props.listenTo(
-            this.propertyModel,
+            this.state.propertyModel,
             'change:value',
             this.onChange.bind(this)
           )
         }
       }
+      componentWillReceiveProps = (nextProps: Props) => {
+        if (nextProps.value !== this.props.value) {
+          this.state.propertyModel &&
+            this.state.propertyModel.setValue([nextProps.value])
+        }
+      }
       onChange() {
         if (this.props.onChange !== undefined) {
-          this.props.onChange(this.propertyModel.getValue()[0])
+          this.props.onChange(this.state.propertyModel.getValue()[0])
         }
       }
       render() {
+        //Forces re-render otherwise the region doesn't get replaced because the view doesn't change
+        const Container = class extends MarionetteRegionContainer {}
```

Comment:
Can you explain how this line forces a re-render? How is this different than using the MarionetteRegionContainer?

Reply:
React will not re-render components if they had't changed and this creates a new instance every time (changes every time). `MarionetteRegionContainer` only picks up changes to the view: https://github.com/codice/ddf/blob/1c555a07bf08654132823fa9f9a69c207aca4a4d/ui/packages/catalog-ui-search/src/main/webapp/react-component/marionette-region-container/marionette-region-container.tsx#L70 Unfortunately `PropertyView` does not result in a re-render on the property value change.

Reply:
You can accomplish the same thing (forcing a rerender each time) by setting a key on the MarionetteRegionContainer like Math.random().